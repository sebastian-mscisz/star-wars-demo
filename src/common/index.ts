export * from './endpoints';
export * from './paths';
export * from './types';
export * from './utils';
