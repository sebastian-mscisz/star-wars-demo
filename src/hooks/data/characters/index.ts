export * from './useCharacters';
export * from './useCharacterDetails';
