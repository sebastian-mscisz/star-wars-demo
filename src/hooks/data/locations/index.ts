export * from './useLocationDetails';
export * from './useLocations';
