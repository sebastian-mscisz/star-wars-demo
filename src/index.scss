body {
    min-height: 100vh;
    display: flex;
    flex-direction: column;
    padding: 0;
    margin: 0;
    background-image: url('/wallpaper.jpg');
    background-repeat: no-repeat;
    background-color: black;
    width: 100%;
    font-family: sans-serif;
}

#root {
    display: flex;
    flex-wrap: wrap;
    flex-grow: 1;
}
