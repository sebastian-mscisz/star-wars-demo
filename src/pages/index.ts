export * from './character-details-page';
export * from './characters-page';
export * from './location-details-page';
export * from './locations-page';
export * from './master-page';
export * from './not-found-page';
export * from './vehicle-details-page';
export * from './vehicles-page';
