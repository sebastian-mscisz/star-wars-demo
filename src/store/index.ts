export * from './data-selectors';
export * from './slices';
export * from './utility';
export * from './rootReducer';
export * from './store';
