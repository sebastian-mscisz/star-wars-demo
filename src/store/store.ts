import { configureStore, EnhancedStore } from '@reduxjs/toolkit';

import rootReducer from './rootReducer';

// eslint-disable-next-line @typescript-eslint/no-explicit-any
export const createStore = (): EnhancedStore<any, any> => {
    const store = configureStore({
        reducer: rootReducer,
    });

    return store;
};

export const store = createStore();

export type AppDispatch = typeof store.dispatch;
