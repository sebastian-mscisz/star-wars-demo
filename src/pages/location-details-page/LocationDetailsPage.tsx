import { LocationDetails } from 'src/components';
import { useRequiredParams } from 'src/common';

export const LocationDetailsPage: React.FC = () => {
    const { locationId } = useRequiredParams<{ locationId: string }>();

    return <LocationDetails locationId={locationId} />;
};
