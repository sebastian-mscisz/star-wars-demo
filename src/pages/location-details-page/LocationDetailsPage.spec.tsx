import { MemoryRouter, Route, Routes } from 'react-router-dom';
import { LocationDetailsPage } from './LocationDetailsPage';
import { screen, render } from '@testing-library/react';
import MockAdapter from 'axios-mock-adapter';
import { DataApi } from 'src/xhr';
import { getCharactersEndpoint, getLocationsEndpoint, locationDetailsPagePath } from 'src/common';
import { Provider } from 'react-redux';
import { createStore } from '../../store';
import { TestCharacters } from 'src/components/characters/test-data';
import { TestLocations } from 'src/components/locations/test-data';

describe('LocationDetailsPage', () => {
    let dataApiMock: MockAdapter;

    beforeEach(() => {
        dataApiMock = new MockAdapter(DataApi);
    });

    it('should render', async () => {
        dataApiMock.onGet(getCharactersEndpoint()).reply(200, TestCharacters);
        dataApiMock.onGet(getLocationsEndpoint()).reply(200, TestLocations);

        const { asFragment } = render(
            <Provider store={createStore()}>
                <MemoryRouter initialEntries={[`${locationDetailsPagePath('2')}`]}>
                    <Routes>
                        <Route path="/location/:locationId" element={<LocationDetailsPage />} />
                    </Routes>
                </MemoryRouter>
            </Provider>,
        );

        await screen.findByText('Location2');

        expect(asFragment()).toMatchSnapshot();
    });
});
