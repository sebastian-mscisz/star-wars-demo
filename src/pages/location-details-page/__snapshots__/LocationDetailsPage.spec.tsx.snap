// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LocationDetailsPage should render 1`] = `
<DocumentFragment>
  <div
    class="sw-location-detail"
  >
    <div
      class="sw-location-detail__name"
    >
      <h2>
        Location2
      </h2>
    </div>
    <div
      class="sw-location-detail__image-wrapper"
    >
      <img
        alt="Star wars location"
        src="/locations/2.jpg"
      />
    </div>
    <div
      class="sw-location-detail__details"
    >
      <p>
        <strong>
          Population:
        </strong>
         5343
      </p>
      <hr />
      <p>
        <strong>
          Residents:
        </strong>
         
        <a
          class="sw-location-detail__details__link"
          href="/character/2"
        >
          Character2
        </a>
      </p>
    </div>
  </div>
  <div
    class="sw-pagination-section sw-pagination-section--center"
    data-testid="pagination-section"
  >
    <div
      class="sw-pagination-button "
      data-testid="left-pagination-button"
    >
      ←
    </div>
    <div
      class="sw-pagination-button "
      data-testid="right-pagination-button"
    >
      →
    </div>
  </div>
</DocumentFragment>
`;
